compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -264,10 +264,13 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             NullaryOp(mir::NullOp::SizeOf, ty) => {
                 let ty = self.subst_from_current_frame_and_normalize_erasing_regions(ty);
                 let layout = self.layout_of(ty)?;
-                assert!(
-                    !layout.is_unsized(),
-                    "SizeOf nullary MIR operator called for unsized type"
-                );
+                if layout.is_unsized() {
+                    // FIXME: This should be a span_bug
```

Comment:
Please also reference `#80742` so we have some way to track what's going on here

Reply:
Done.